Sources/TuistCore/Graph/GraphTraverser.swift

```diff
@@ -231,9 +231,12 @@ public class GraphTraverser: GraphTraversing {
         let externalBundles = filterDependencies(
             from: .target(name: name, path: path),
             test: { dependency in
-                isDependencyResourceBundle(dependency: dependency) &&
-                    (isDependencyExternal(dependency) || dependency.isPrecompiled) &&
-                    canEmbedBundles(target: target)
+                guard isDependencyResourceBundle(dependency: dependency) else { return false }
+                // Precompiled bundles are embedded to any downstream target that supports resources to ensure Xcode previews work
```

Comment:
In a graph like `A -(depens on)->  B -> C`, would you consider going from C to A upstream or downstream? I would say going upstream there, but I think we are looking at the graph differently.

Reply:
From C to A should be downstream, no? C needs to be built _before_ A, so I'd consider C to be upstream from A.